source/mir/ndslice/topology.d

```diff
@@ -3067,3 +3069,411 @@ unittest
     assert(lowerData == [0, 3, 4, 6, 7, 8]);
     assert(upperData == [0, 1, 2, 4, 5, 8]);
 }
+
+// Transposed adjusted to ignore dim=0 and include universal
+private template adjTransposed(size_t dim = 0)
+{
+    import mir.ndslice : Slice, SliceKind;
+	
+    auto adjTransposed(SliceKind kind, size_t[] packs, Iterator)(Slice!(kind, packs, Iterator) slice)
+    {
+        static assert (packs != [1], "1-dimensional slice not accepted");
+		
+        static if (dim == 0)
+        {
+            return slice;
+        }
+        else
+        {
+            import mir.ndslice : universal, transposed;
+            return slice.universal.transposed!dim;
+        }
+    }
+}
+
+// 2-dimensional slice support
+@safe @nogc pure nothrow
+unittest
+{
+    import mir.ndslice : iota;
+	
+    auto slice = iota(3, 4);
+	
+    static immutable shape34 = [3, 4];
```

Comment:
Please change to fixed size arrays (like `size_t[2] = shape34 = [3, 4];` ) for all unittests